lib/topological_inventory/openshift/collectors_pool.rb

```diff
@@ -0,0 +1,85 @@
+require "config"
+require "topological_inventory/openshift/collector"
+require "topological_inventory/openshift/logging"
+
+module TopologicalInventory::Openshift
+  class CollectorsPool
+    include Logging
+
+    def initialize(config_name, metrics, poll_time: 10)
+      self.collectors  = {}
+      self.collector_threads = {}
+      self.config_name = config_name
+      self.metrics     = metrics
+      self.poll_time   = poll_time
+    end
+
+    def run!
+      loop do
+        reload_config
+
+        remove_old_collectors
+        add_new_collectors
+
+        sleep(poll_time)
+      end
+    end
+
+    def stop!
+      collectors.each_value { |collector| collector.stop }
+
+      # Wait for end of collectors to ensure metrics are stopped after them
+      collector_threads.each { |thread| thread.join }
+    end
+
+    private
+
+    attr_accessor :collectors, :collector_threads, :config_name, :metrics, :poll_time
+
+    def reload_config
+      clear_settings
+
+      config_file = File.join(path_to_config, "#{sanitize_filename(config_name)}.yml")
+      raise "Configuration file #{config_name} doesn't exist" unless File.exist?(config_file)
+
+      ::Config.load_and_set_settings(config_file)
+    end
+
+    def add_new_collectors
+      (::Settings.sources || []).each do |source|
```

Comment:
Usually `Array(Settings.sources)` or `Settings.sources.to_a` is used over `|| []`, just a bit safer